vars/liskBuild.groovy

```diff
@@ -0,0 +1,47 @@
+
+
+def call(Map params) {
+	def liskBranch = params.branch
+	def buildNetwork = params.network
+
+	def liskDir = "lisk-${liskBranch}"
+	def liskVersion = ""
+	def liskReleaseName = ""
+
+	def liskBuildDir = ""
+	def liskBuildBranch = ""
+	def	latestVersion = false 
+
+	dir(liskDir) {
+		git url: "https://github.com/LiskHQ/lisk.git", branch: liskBranch
+
+		sh """
+		npm install
+		node ./node_modules/.bin/grunt release
+		jq -r '.version' config.json > ".lisk-version"
+		"""
+	}
+	script {
+		liskVersion = readFile("${liskDir}/.lisk-version").trim()
+
+		if(fileExists("${liskDir}/genesisBlock.json")) {
+			liskBuildBranch = "0.9.15"
+			liskReleaseName = "${liskVersion}.tar.gz"
+			latestVersion = false
+		} else {
+			liskBuildBranch = "1.0.0"
+			liskReleaseName = "lisk-${liskVersion}.tgz"
+			latestVersion = true
+		}
+		liskBuildDir = "lisk-build-${liskBuildBranch}"
+	}
+	dir(liskBuildDir) {
+		git url: "https://github.com/LiskHQ/lisk-build.git", branch: liskBuildBranch
+		sh """
+		cp "../${liskDir}/release/${liskReleaseName}" "src/"
+		bash build.sh -n "${buildNetwork}" -v "${liskVersion}"
+		"""
+	}
+
+	return [version: liskVersion, latestVersion: latestVersion, network: buildNetwork, file: "${liskBuildDir}/release/lisk-${liskVersion}-Linux-x86_64.tar.gz"]
```

Comment:
`buildNetwork` is set from `params.network`: why do we return a parameter?

Reply:
For the sake of consistent and valuable response.

Reply:
So... I pass a parameter and get it back? Should I check that it's `==` what I passed? This is a little confusing.

Reply:
You didn't get the same parameter back. You get a response, 

```
def response = liskBuild(network: 'dev', branch: 'master') 
```

Now that `response` object contains all information that on which branch, network the build was called, what version of lisk it was and where is the file that it builds. 

```
response.version
response.network
response.latestVersion # 1.0.0 > 
response.file
```

I don't think there is something to get confused. Had you never tried calling any HTTP API with some parameter, and get response containing same parameter in the meta data to make response more useful and informative.